lib/url-parse.js

```diff
@@ -14,36 +13,38 @@ var url = require('url')
 percentEncode = function percentEncode(c) {
   var hex = c.toString(16).toUpperCase();
   (hex.length === 1) && (hex = '0' + hex)
-  return '%' + hex
+  return PERCENT + hex
 }
 
 
 /**
- * Percent encodes a query string according to RFC 3986
+ * Percent partialEncode a query string according to RFC 3986
  *
  * @param value
  * @returns {string}
  */
-encode = function (value) {
+partialEncode = function (value) {
   if (!value) { return '' }
 
   var buffer = new Buffer(value),
     ret = '',
     i
 
   for (i = 0; i < buffer.length; ++i) {
-    if (buffer[i] < 0x21 || buffer[i] > 0x7E || buffer[i] === 0x22 || buffer[i] === 0x23 ||
```

Comment:
See this document it says clearly the steps.
https://url.spec.whatwg.org/#parsers

When we find % in URL, we need to check if next two characters or next four characters are hex or not. If not, then we encode the % to %25

```
Let output be an empty byte sequence.

For each byte byte in input, run these steps:

If byte is not `%`, append byte to output.

Otherwise, if byte is `%` and the next two bytes after byte in input are not in the ranges 0x30 to 0x39, 0x41 to 0x46, and 0x61 to 0x66, append byte to output.

Otherwise, run these substeps:

Let bytePoint be the two bytes after byte in input, decoded, and then interpreted as hexadecimal number.

Append a byte whose value is bytePoint to output.

Skip the next two bytes in input.
```